src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -695,6 +715,28 @@ protected virtual void Dispose(bool finalize)
 
         // We're disposing this workspace.  Stop any work to update SG docs in the background.
         _updateSourceGeneratorsQueueTokenSource.Cancel();
+        _workQueueTokenSource.Cancel();
+    }
+
+    private async ValueTask ProcessWorkQueueAsync(ImmutableSegmentedList<Action> list, CancellationToken cancellationToken)
+    {
+        // Hop over to the right scheduler to execute all this work.
+        await Task.Factory.StartNew(() =>
+        {
+            foreach (var item in list)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    item();
+                }
+                catch (Exception e) when (FatalError.ReportAndCatchUnlessCanceled(e))
```

Comment:
Not sure why the "UnlessCancelled" here -- I wouldn't expect cancellation from the action since we never gave it a token.